Implemented movie details UI

LeonardoMolleker opened the pull request:
![ezgif com-gif-maker (9)](https://user-images.githubusercontent.com/84337338/122589165-4f2b9600-d036-11eb-9478-075637ce9353.gif)

A page that displays the details of a movie.

LeonardoMolleker committed:
Implemented movie details UI

LeonardoMolleker committed:
Added hero transition

FranciscoGaraventa commented on lib/constants/string_constants.dart:
```diff
@@ -11,4 +11,9 @@ abstract class StringConstants {
+  static const String movieDetailRating = " - Rating: ";
+  static const String playButtonText = "Play";
+  static const String heroMovieDetailTransitionTag = "movieImage";
+}
```
You missed a newline at end of file.

LeonardoMolleker replied:
Done.

FranciscoGaraventa commented on lib/pages/movie_details_page.dart:
```diff
@@ -0,0 +1,139 @@
+                  MeasuresConstants.movieDetailOverviewLeftRightBottomPadding,
+              bottom:
+                  MeasuresConstants.movieDetailOverviewLeftRightBottomPadding,
+            ),
+            child: Text(
+              result.overview,
+              style: TextStyle(
+                color: Colors.white,
+                fontSize: MeasuresConstants.movieDetailOverviewFontSize,
+              ),
+            ),
+          ),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Icon(
+                Icons.share_outlined,
+                color: Colors.white,
+              ),
+              SizedBox(
+                width: MeasuresConstants.movieDetailSixedBoxWidth,
+              ),
+              Icon(
+                Icons.favorite_border,
+                color: Colors.white,
+              ),
+              SizedBox(
+                width: MeasuresConstants.movieDetailSixedBoxWidth,
+              ),
+              Icon(
+                Icons.add,
+                color: Colors.white,
+              ),
+            ],
+          )
+        ],
+      ),
+    );
+  }
+}
```
Can you separate this into different widgets like: BackdropImage, Overview, Movie, some widget that has language, date and rating.

FranciscoGaraventa replied:
Add the last one that contains Icons of the bottom.

LeonardoMolleker replied:
Done.

LeonardoMolleker committed:
abstracted widgets of movie detail page. Update: exported movie
widgets. Fix update: changed constants names, changed parameters of
getImage function, hero widget is now in MovieDetailImage and
HombeBodyContent.

FranciscoGaraventa commented on lib/library/movies_widgets_package.dart:
```diff
@@ -0,0 +1,7 @@
+export '../widgets/movie_details_actions.dart';
+export '../widgets/movie_details_info.dart';
+export '../widgets/movie_details_overview.dart';
+export '../widgets/movie_details_title.dart';
```
You missed a newline at end of file here

LeonardoMolleker replied:
Added

FranciscoGaraventa commented on lib/pages/movie_details_page.dart:
```diff
@@ -0,0 +1,49 @@
+      body: SingleChildScrollView(
+        child: Column(
+          children: [
+            MovieDetailImage(
+              image: functionGetImage(
+                result,
+              ),
+            ),
```
Instead of send a widget, what do you think if you only have two parameter, the first one is for the posterPath and the second one relative to a default image path

FranciscoGaraventa replied:
You can add another parameter like tag, like i see you use it in Hero widget

LeonardoMolleker replied:
Changed

FranciscoGaraventa commented on lib/widgets/movie_detail_image.dart:
```diff
@@ -0,0 +1,25 @@
+        Container(
+          height: MeasuresConstants.movieDetailContainerHeight,
+          child: Center(
+            child: image,
```
Here you can add the Hero widget that you create in home_body_content

LeonardoMolleker replied:
Added

FranciscoGaraventa commented on lib/widgets/movie_detail_play_button.dart:
```diff
@@ -0,0 +1,45 @@
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(
+        left: MeasuresConstants.playButtonLeftRightPadding,
+        right: MeasuresConstants.playButtonLeftRightPadding,
+      ),
```
You can change this from EdgeInsets.only to EdgeInsets.symmetric and change left/right with horizontal

LeonardoMolleker replied:
Changed

FranciscoGaraventa commented on lib/widgets/movie_details_overview.dart:
```diff
@@ -0,0 +1,30 @@
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(
```
This can be EdgeInsets.LTRB.

LeonardoMolleker replied:
Changed

FranciscoGaraventa commented on lib/pages/movie_details_page.dart:
```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:tmdb_movies/constants/string_constants.dart';
```
You missed a relative import here

LeonardoMolleker replied:
Done.

FranciscoGaraventa commented on lib/widgets/home_body_content.dart:
```diff
@@ -42,21 +44,50 @@ class HomeBodyContent extends StatelessWidget {
     });
   }
+
+  FadeInImage getImage({
+    String posterPath,
+    String defaultPath = StringConstants.defaultPoster,
+    String id,
+  }) {
+    return FadeInImage.assetNetwork(
+      placeholder: defaultPath,
+      image: posterPath != null
+          ? StringConstants.uriPosterImage + posterPath
+          : defaultPath,
+      fit: BoxFit.cover,
+    );
+  }
```
You are using in two different places, here in home_body_content and in movie_detail_image and basically is the same code so you can create a new widget for this.

LeonardoMolleker replied:
Done.

LeonardoMolleker committed:
deleted library directory, abstracted fadeIn widget

MarcosCGlobant reviewed (approved):
LGTM

FranciscoGaraventa reviewed (approved):
LGTM
